syntax = "proto3";

message Ticket {
    int32 leg_id = 1;
    string departure_place = 2;
    string arrival_place = 3;
    string fligh_date = 4;
    float total_fare = 5;
}

message Airline {
    int32 airline_code = 1;
    string airline_name = 2;
}

message TicketsRequest {
    string departure_place = 1;
    string arrival_place = 2;
}

message TicketsResponse {
    repeated Ticket tickets = 1;
}

message AirlineRequest {
    string airline_code = 1;
}

message AirlineResponse {
    Airline airline = 1;
}

service Visualization {
    rpc GetTickets (TicketsRequest) returns (TicketsResponse);
    rpc GetAirline (AirlineRequest) returns (AirlineResponse);
}
